frontend/src/workspace/buildingtools/FunNode.js

```diff
@@ -52,20 +52,21 @@
 // | Notes |
 // +-------+---------------------------------------------------------
 
-
 // +----------------------------+------------------------------------
 // | All dependent files        |
 // +----------------------------+
 
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect, useRef, useContext } from "react";
 import { Rect, Group, Text, Shape, Image } from "react-konva";
 import Konva from "konva";
 import Portal from "./Portal";
-import gui from "./mistgui-globals.js";
+import gui from "../globals/mistgui-globals.js";
 import MISTImage from "./MISTImage";
 import useImage from "use-image";
-import nodeDimensions from "./globals-nodes-dimensions.js";
-import globals from "./globals.js";
+import { nodeContext } from "../globals/globals-nodes-dimensions.js";
+import { globalContext } from "../globals/global-context";
+import globals from "../globals/globals.js";
```

Comment:
Since you don't seem to use Portal, MISTImage, or globals, you should remove them.